internal/common/metrics/grpc.go

```diff
@@ -59,9 +59,17 @@ func NewGRPCInterceptor(defaultHandler Handler, suffix string) grpc.UnaryClientI
 
 		// Only take method name after the last slash
 		operation := method[strings.LastIndex(method, "/")+1:]
-		handler = handler.WithTags(map[string]string{OperationTagName: operation})
+		tags := map[string]string{OperationTagName: operation}
+
+		// Since this interceptor can be used for clients of different name, we
+		// attempt to extract the namespace out of the request. All namespace-based
+		// requests have been confirmed to have a top-level namespace field.
```

Comment:
Does this deserve a comment in the proto file? I mean I wouldn't do a PR just for that, but bundle it into your next api PR? :)

Reply:
Yeah, if I remember on my next one I'll add something like "Some clients expect all namespace-based RPC calls to have a single `namespace` string field in the request".